src/main/java/com/amazon/ion/impl/schubfach/DoubleToDecimal.java

```diff
@@ -0,0 +1,673 @@
+     * <i>d</i><sub>1</sub>&hellip;<i>d</i><sub><i>n</i></sub><!--
+     * --><code>0</code>&hellip;<code>0.0</code>,
+     * where there are exactly <i>i</i> zeroes
+     * between <i>d</i><sub><i>n</i></sub> and the decimal point.
+     * For example, 123 &times; 10<sup>2</sup> is formatted as
+     * {@code 12300.0}.
+     * <li>Subcase <i>i</i> &lt; 0:
+     * <i>d</i><sub><code>v</code></sub> is formatted as
+     * <i>d</i><sub>1</sub>&hellip;<!--
+     * --><i>d</i><sub><i>n</i>+<i>i</i></sub>.<!--
+     * --><i>d</i><sub><i>n</i>+<i>i</i>+1</sub>&hellip;<!--
+     * --><i>d</i><sub><i>n</i></sub>.
+     * There are exactly -<i>i</i> digits to the right of
+     * the decimal point.
+     * For example, 123 &times; 10<sup>-1</sup> is formatted as
+     * {@code 12.3}.
+     * </ul>
+     * <li>Case <i>e</i> &lt; -3 or <i>e</i> &ge; 7:
+     * computerized scientific notation is used to format
+     * <i>d</i><sub><code>v</code></sub>.
+     * Here <i>e</i> is formatted as by {@link Integer#toString(int)}.
+     * <ul>
+     * <li>Subcase <i>n</i> = 1:
+     * <i>d</i><sub><code>v</code></sub> is formatted as
+     * <i>d</i><sub>1</sub><code>.0E</code><i>e</i>.
+     * For example, 1 &times; 10<sup>23</sup> is formatted as
+     * {@code 1.0E23}.
+     * <li>Subcase <i>n</i> &gt; 1:
+     * <i>d</i><sub><code>v</code></sub> is formatted as
+     * <i>d</i><sub>1</sub><code>.</code><i>d</i><sub>2</sub><!--
+     * -->&hellip;<i>d</i><sub><i>n</i></sub><code>E</code><i>e</i>.
+     * For example, 123 &times; 10<sup>-21</sup> is formatted as
+     * {@code 1.23E-19}.
+     * </ul>
+     * </ul>
+     *
+     * @param v the {@code double} to be rendered.
+     * @return a string rendering of the argument.
+     */
+    public static String toString(double v) {
+        return threadLocalInstance().toDecimalString(v);
+    }
+
+    /**
+     * Appends the rendering of the {@code v} to {@code app}.
+     *
+     * <p>The outcome is the same as if {@code v} were first
+     * {@link #toString(double) rendered} and the resulting string were then
+     * {@link Appendable#append(CharSequence) appended} to {@code app}.
+     *
+     * @param v the {@code double} whose rendering is appended.
+     * @param app the {@link Appendable} to append to.
+     * @throws IOException If an I/O error occurs
+     */
+    public static Appendable appendTo(double v, Appendable app)
+            throws IOException {
+        return threadLocalInstance().appendDecimalTo(v, app);
+    }
+
+    private static DoubleToDecimal threadLocalInstance() {
+        return threadLocal.get();
+    }
+
+    private String toDecimalString(double v) {
+        switch (toDecimal(v)) {
+            case NON_SPECIAL: return charsToString();
+            case PLUS_ZERO: return "0e0";
+            case MINUS_ZERO: return "-0e0";
+            case PLUS_INF: return "Infinity";
+            case MINUS_INF: return "-Infinity";
+            default: return "NaN";
```

Comment:
See [Ion Floats: Special Values](https://amazon-ion.github.io/ion-docs/docs/float.html#special-values).

```suggestion
            case PLUS_INF: return "+inf";
            case MINUS_INF: return "-inf";
            default: return "nan";
```